aspnet/mvc/views/forms/sample/final/Views/Demo/EditColor.cshtml

```diff
@@ -0,0 +1,17 @@
+﻿@model Person
+@{
+    var index = (int)ViewData["index"];
+}
+
+<form asp-controller="ToDo" asp-action="Edit" method="post">
+    @*
+        Move the following two lines to a template to avoid
+        evaluating index twice.
+        <label asp-for="Colors[index]" />
```

Comment:
Never use a self-closing `<label>`. This will not work correctly since the generated HTML will also contain an invalid element. Search the sources because this error appears elsewhere as well. (The only HTML elements used in these samples that should be written in self-closing form are `<input/>` and `<br/>`.)